ecologits/impacts/modeling.py

```diff
@@ -22,8 +25,8 @@ class Impact(BaseModel):
     value: ValueOrRange
     unit: str
 
-    def __add__(self, other: "Impact") -> "Impact":
-        if not isinstance(other, Impact):
+    def __add__(self: Impact, other: "BaseImpact") -> Impact:
+        if not isinstance(other, BaseImpact):
```

Comment:
question: as the type is checked and an error raised, we could just have a more generic "object" type?
I think I had a "not compliant with Liskov principles" on this one from mypy

Reply:
Yes I agree it makes more sense if we compare to comparison methods.